catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/MigrationTaskManager.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    /**
+     * Must be called after exportSetup and all calls to exportMetacardQuery. This will attempt
+     * to gracefully shutdown the executor service and handle any exceptions that might have
+     * occured in the asynchronous threads.
+     * <p>
+     * If you are using {@link MigrationTaskManager}; that is, you have constructed an instance,
+     * you must call this method to gracefully release resources and shutdown.
+     *
+     * @throws MigrationException thrown if some of the some of the metacards couldn't be exported
+     */
+    public void exportFinish() throws MigrationException {
+        LOGGER.debug("Attempting to shutdown catalog export");
+        taskExecutor.shutdown();
+        boolean isGracefulTermination = false;
+        try {
+            isGracefulTermination = taskExecutor.awaitTermination(1L, TimeUnit.MINUTES);
+        } catch (InterruptedException e) {
+            LOGGER.warn("Executor was interrupted during shutdown: " + e.getMessage(), e);
+        } finally {
+            if (!isGracefulTermination) {
+                taskExecutor.shutdownNow();
+            }
+
+            checkForFailures();
+        }
+    }
+
+    /**
+     * Creates a sublist from the given results and a destination file for metacards based on
+     * the {@link CatalogMigratableConfig} object used by this task manager. An asynchronous
+     * task is started for exporting the metacards and is deferred to the appropriate instance
+     * of {@link MigrationFileWriter} for processing.
+     *
+     * @param results          The results of a catalog query that need to be exported.
+     * @param exportGroupCount The group or page number of the query. This value is not monitored
+     *                         or audited for repeats and is strictly for record keeping by naming
+     *                         the resulting export files appropriately.
+     * @throws MigrationException Thrown if one of the writing threads fails or throws an exception
+     *                            itself.
+     */
+    public void exportMetacardQuery(final List<Result> results, long exportGroupCount)
+            throws MigrationException {
+        for (int i = 0; i < results.size(); i += catalogConfig.getExportCardsPerFile()) {
+            final List<Result> fileResults = results.subList(i,
+                    Math.min((i + catalogConfig.getExportCardsPerFile()), results.size()));
+            final File exportFile = catalogConfig.getExportPath()
+                    .resolve(makeFileName(exportGroupCount, i))
+                    .toFile();
+
+            CatalogWriterCallable writerCallable = createWriterCallable(exportFile,
+                    fileResults,
+                    fileWriter);
+
+            ListenableFuture<Void> task = taskExecutor.submit(writerCallable);
+
+            Futures.addCallback(task, createFutureCallback());
+        }
+
+        checkForFailures();
+    }
+
+    /**
```

Comment:
We should use a single in-line comment on the methods that are package-private instead of using a block comment, otherwise we may end up with a comment that doesn't follow the method(s) in question and is inaccurate. For instance in this case, `getAtomicReference()` is not overwritten anymore and can be `private`.